astropy/io/fits/_utils.pyx

```diff
@@ -0,0 +1,57 @@
+from collections import OrderedDict
+
+cdef Py_ssize_t BLOCK_SIZE = 2880  # the FITS block size
+cdef Py_ssize_t CARD_LENGTH = 80
+cdef str VALUE_INDICATOR = '= '  # The standard FITS value indicator
+cdef str END_CARD = 'END' + ' ' * 77
+
+
+def parse_header(fileobj):
+    """The main method allowing to parse quickly a FITS header."""
+
+    cards = OrderedDict()
+    read_blocks = []
+    cdef int found_end = 0
+    cdef bytes block
+    cdef str header_str, block_str, card_image, keyword
+    cdef Py_ssize_t idx, end_idx, sep_idx
+
+    while True:
```

Comment:
Wouldn't it make sense to use `while found_end == 0` here?

Reply:
Sure!